src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -274,15 +275,27 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
     // means that we don't need an additional field on the work in
     // progress.
     const current = workInProgress.alternate;
-    const next = beginWork(current, workInProgress, nextPriorityLevel);
 
-    if (next) {
-      // If this spawns new work, do that next.
-      return next;
-    } else {
-      // Otherwise, complete the current work.
-      return completeUnitOfWork(workInProgress);
+    if (debugTool) {
```

Comment:
We should wrap all of these in `__DEV__` before we have the infrastructure in place to statically DCE them.